ERROR: Abort in frontends/ast/simplify.cc:4239.

## Steps to reproduce the issue

Current version of yosys reproduces the problem in the title with both ice40 and ecp5 targets. The error message is shown pretty much immediately.

```
git clone https://github.com/Speccery/icy99.git
make

yosys  -q -p "synth_ice40 -top top_blackice2 -abc2 -blif erik9900.blif" src/ram2.v src/sys.v src/rom.v src/tms9900.v src/alu9900.v src/tms9902.v src/erik_pll.v src/tms9918.v src/vga_sync.v src/dualport1k9.v src/dualport4k.v src/dualport512_9.v src/xmemctrl.v src/gromext.v src/serloader.v src/serial_rx.v src/serial_tx.v src/spi_slave.v src/tms9901.v src/lcd_sys.v lcd/pmodoledrgb_controller.v lcd/ram_source.v src/dualport_par.v src/ps2kb.v top_blackice2.v
Warning: Replacing memory \keyboard with list of registers. See src/sys.v:374, src/sys.v:361, src/sys.v:343
ERROR: Abort in frontends/ast/simplify.cc:4239.
Makefile:18: recipe for target 'erik9900.blif' failed
make: *** [erik9900.blif] Error 1
```

Sorry that I haven't minimized the code (yet). Will try to reduce code size later, but without context where the error originates it's a bit hard to know where to start reducing the code. The warning above occurs in my current code. 
Build from 2020-05-24 works, current version consistently produces the error above.

## Expected behavior
Build proceeds.
The build will stop prematurely as my git repository does not included the ROM memory initialization files currently (on purpose). 

## Actual behavior
**ERROR: Abort in frontends/ast/simplify.cc:4239.**

The error above is produced almost immediately.

--- a/frontends/ast/simplify.cc
+++ b/frontends/ast/simplify.cc
@@ -4231,6 +4231,8 @@ bool AstNode::detect_latch(const std::string &var)
 			case AST_POSEDGE:
 			case AST_NEGEDGE:
 				return false;
+			case AST_EDGE:
+				break;
 			case AST_BLOCK:
 				if (!c->detect_latch(var))
 					return false;

